File: leetcode/addbinary/addbinary.c
char * addBinary(char * a, char * b){
    int lena = strlen(a);
    int lenb = strlen(b);
    char *aptr = a + lena - 1;
    char *bptr = b + lenb - 1;
    char *result = NULL;
    int maxresult = (lena > lenb ? lena : lenb) + 2;
    int carry = 0;
    result = malloc(sizeof(char)*maxresult);
    
    result[maxresult - 1] = '\0';
    maxresult--;
    while (aptr >= a || bptr >= b) {
        char s[2] = {'\0'};
        int val = 0;
        if (aptr >= a) {
            s[0] = *aptr;
            val += atoi(s);
            aptr--;
        }
        if (bptr >= b) {
            s[0] = *bptr;
            val += atoi(s);
            bptr--;
        }
        if (carry) {
            val++;
            carry = 0;
        }
        if (val > 1) {
            carry = 1;
            val -= 2;
        }
        result[maxresult - 1] = val == 0 ? '0' : '1';
        maxresult--;
    }
    if (carry) {
        result[maxresult - 1] = '1';
    } else
        result++;
    
    return result;
    
}

File: leetcode/addbinary/addbinary.py
class Solution(object):
    def addBinary(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        result = list()
        lena = len(a)
        lenb = len(b)
        carry = False
        while lena + lenb > 0:
            vala = 0
            valb = 0
            if lena > 0:
                vala = int(a[lena - 1])
                lena -= 1
            if lenb > 0:
                valb = int(b[lenb - 1])
                lenb -= 1
            val = vala + valb
            if carry:
                val += 1
                carry = False
            if val > 1:
                carry = True
                val -= 2
            result.insert(0, str(val))
        if carry:
            result.insert(0, '1')
        s = ""
        return s.join(result)
